### DISCUSSION TITLE
Can I instruct p2 to use only the specified URL and no mirrors

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-equinox/p2#269] [General]
A client of mine uses a proxy which blocks certain servers. Installing new / additional features from the release update sites as https://download.eclipse.org/releases/2023-03 via the Update manager from the Eclipse IDE is hard for him, as the p2 update UI also uses mirrors and these are blocked for him.

![image](https://github.com/eclipse-equinox/p2/assets/139910/9bcef0fd-a3f2-4b38-a6ec-0f0858b7fc7b)

Can I somehow control that p2 does not use proxies for the installation?

cc @merks as Ed works a lot with p2.

**Comment by merks** [comment]
This static constant

https://github.com/eclipse-equinox/p2/blob/aa4840ff474976da781a066cabb4eb5029f2790c/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java#L62

is use here to guard fetching the mirror URL:

https://github.com/eclipse-equinox/p2/blob/aa4840ff474976da781a066cabb4eb5029f2790c/bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java#L755-L762

So `-Declipse.p2.mirrors=false` should do the trick.
